### ISSUE TITLE
[bug][aarch64] #pragma pack() and volatile keyword will get wrong result

### ISSUE THREAD

**Issue by MLsheng** [llvm/llvm-project#64580]
My compiler version: Clang Arm64BE 15.0.4
```cpp
#include <stdint.h>
#include <stdio.h>

#pragma pack(1)
struct {
    unsigned a : 18;
    unsigned b : 24;
    volatile unsigned c : 27;
    unsigned : 20;
    signed : 10;
    signed : 15;
    unsigned : 13;
} d[] = {{0,0,0x7ffffff}};

int main() {
    printf("%x\n", d[0].c);
    return 0;
}
```

##### Problem

With volatile declare variable `c`, the output is `0`, wrong;
without volatile, the output is `7ffffff`, right.

##### Assembly

Memory layout of the struct

```asm
d:
	.byte	0                               // 0x0
	.byte	0                               // 0x0
	.byte	0                               // 0x0
	.byte	0                               // 0x0
	.byte	0                               // 0x0
	.byte	63                              // 0x3f
	.byte	255                             // 0xff
	.byte	255                             // 0xff
	.byte	248                             // 0xf8
	.zero	7
	.size	d, 16
```

with volatile：

```asm
adrp	x8, d
add	x8, x8, :lo12:d
ldp	x8, x9, [x8]
lsr	x8, x8, #59
bfi	w8, w9, #5, #27
and	w1, w8, #0x7ffffff
adrp	x0, .L.str
add	x0, x0, :lo12:.L.str
bl	printf
```

without volatile：

```asm
adrp	x8, d+4
add	x8, x8, :lo12:d+4
ldr	w9, [x8]
ldur	x8, [x8, #4]
lsr	x8, x8, #59
bfi	w8, w9, #5, #27
and	w1, w8, #0x7ffffff
adrp	x0, .L.str
add	x0, x0, :lo12:.L.str
bl	printf
```
I guess that the register of volatile variable is wrong.

**Comment by MLsheng** [comment]
Does anyone have ideas on this issue?

**Comment by davemgreen** [comment]
I think the problem is that i128 volatile loads are the wrong way around under BE. Here is a simpler example: https://godbolt.org/z/cKbx5816j

The results of the AArch64ISD::LDP probably need to be the other way around in this merge: https://github.com/llvm/llvm-project/blob/33df617dfda3298730ccadb44c69ccffe84c8716/llvm/lib/Target/AArch64/AArch64ISelLowering.cpp#L24187

**Comment by MLsheng** [comment]
> I think the problem is that i128 volatile loads are the wrong way around under BE. Here is a simpler example: https://godbolt.org/z/cKbx5816j
> 
> The results of the AArch64ISD::LDP probably need to be the other way around in this merge:
> 
> https://github.com/llvm/llvm-project/blob/33df617dfda3298730ccadb44c69ccffe84c8716/llvm/lib/Target/AArch64/AArch64ISelLowering.cpp#L24187

thank you!

**Comment by davemgreen** [comment]
There is hopefully a fix in #67413
